tensorboard/uploader/uploader.py

```diff
@@ -503,7 +534,12 @@ def flush(self):
 
         self._rpc_rate_limiter.tick()
 
-        with _request_logger(request, request.runs):
+        with contextlib.ExitStack() as stack:
```

Comment:
Took me a bit to figure out why you did it this way :)  I think it's because `self._tracker` might be `None`, so you can't just `with self._tracker.scalars_tracker(...):`.

I think this would be clearer, more idiomatic, and more flexible if you don't allow `self._tracker` to be None, but instead push the verbosity argument down into UploadTracker and handle it there (i.e., checking the condition only when it comes time to actually print something).  Then all of these ExitStack things go away and you can just say

```
with _request_logger(request, request.runs):
    with self._tracker.scalars_tracker(self._num_values):
        ...
```

Reply:
Done. This increases the maximum depth of ident by 1. But I think it's worth the benefits you pointed out.